.github/workflows/gradle.yml

```diff
@@ -0,0 +1,85 @@
+name: Java CI
+on:
+  push:
+    branches:
+      - '[4-9]+.[0-9]+.x'
+      - '[3-9]+.[3-9]+.x'
+  pull_request:
+    branches:
+      - '[1-9]+.[0-9]+.x'
```

Comment:
I think this should be similar to push to branches.

Reply:
If we set `on.pull_request.branches` to:

```
      - '[4-9]+.[0-9]+.x'
      - '[3-9]+.[3-9]+.x'
```

Then the action will not be triggered for this PR, because it targets branch `3.0.x`. In fact, I don't see the point in having several of these regular expressions, instead of just one catch-all `[1-9]+.[0-9]+.x` (both for `on.pull_request.branches` and `on.push.branches`) 🤔 Is there any particular reason to do things that way @puneetbehl  ?

Reply:
I believe the reason we have multiple regex to skip executing this workflow for older branches.

Reply:
@puneetbehl But this workflow will not be executed against older branches unless we actually merge this PR into them. So I don't think we need to _skip_ anything, really 🤔

In any case, if for some reason we wanted to avoid executing this workflow for versions below `3.0.x`, we could do that with just one regex: `[3-9]+.[0-9]+.x`.